clustered/server/src/main/java/org/ehcache/clustered/server/store/ClusterTierActiveEntity.java

```diff
@@ -112,9 +118,16 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(ClusterTierActiveEntity.class);
   static final String SYNC_DATA_SIZE_PROP = "ehcache.sync.data.size.threshold";
-  private static final long DEFAULT_SYNC_DATA_SIZE_THRESHOLD = 4 * 1024 * 1024;
+  private static final long DEFAULT_SYNC_DATA_SIZE_THRESHOLD = 2 * 1024 * 1024;
+  static final String SYNC_DATA_GETS_PROP = "ehcache.sync.data.gets.threshold";
+  private static final int DEFAULT_SYNC_DATA_GETS_THRESHOLD = 16 * 1024;
+
   static final String CHAIN_COMPACTION_THRESHOLD_PROP = "ehcache.server.chain.compaction.threshold";
-  static final int DEFAULT_CHAIN_COMPACTION_THRESHOLD = 8;
+  private static final int DEFAULT_CHAIN_COMPACTION_THRESHOLD = 8;
+
+  private static final LastDataSyncMessage SYNC_MESSAGE_COMPLETION_MARKER = new LastDataSyncMessage();
+  private static final int MAX_SYNC_CONCURRENCY = 1;
+  private static final Executor SYNC_GETS_EXECUTOR = Executors.newFixedThreadPool(MAX_SYNC_CONCURRENCY);
```

Comment:
This thread pool is going to keep it's threads alive forever.  That seems wasteful, why can't we have the threads alive only when we're passive syncing?

Reply:
@chrisdennis , Currently there is no startSync() interface on the active that tells that a sync is beginning. 

I can create the threadpool when segment 0 is syncing, but that would cause issues if there is concurrent syncing, where I will have to make the threadpool creation synchronous. Pls advise on which is better.

Reply:
I might be naive but why no do this?

```
      ExecutorService executor = Executors.newSingleThreadExecutor();

      try {
        CompletableFuture<Void> asyncGet = CompletableFuture.runAsync(
          () -> doGetsForSync(segmentId, messageQ, Thread.currentThread()), executor);

        while (true) {
          try {
            EhcacheDataSyncMessage msg = messageQ.take();
            if (msg instanceof LastDataSyncMessage) {
              break;
            } else {
              syncChannel.synchronizeToPassive(msg);
            }
          } catch (InterruptedException e) {
            interrupted = true;
            break;
          }
        }

        if (interrupted) {
          // get is finite memory operation..give it time to complete even if async get task is not interrupted
          // for some reason so that we can get the actual cause of exception.
          asyncGet.get(10, TimeUnit.SECONDS);
          // the get threads actual execution status precedes our own interruption status
          throw new InterruptedException();
        } else {
          asyncGet.get();
        }
      } catch (InterruptedException | ExecutionException | TimeoutException e) {
        throw new RuntimeException(e);
      } finally {
        executor.shutdownNow();
      }
    }
```

Reply:
@henri-tremblay This will create a thread executor for every concurrency. .. That is 16 times per concurrency

Terracotta core calls this method 16 times during passive sync..Once for every concurrency..Also I am trying to push for the core to do atleast 4 concurrency in parallel for nearly another 3X additional performance. 

Is it ok to create a thread pool executor for each concurrency and shut it down?

Reply:
Just configure the thread pool to have a finite keep alive time.